torchgen/gen_functionalization_type.py

```diff
@@ -391,12 +401,25 @@ def emit_view_functionalization_body(
         return at::_ops::{noop_api_name}::call({', '.join(view_redispatch_args)});
       }}
       auto reapply_views = at::functionalization::impl::getFunctionalizationReapplyViewsTLS();
-      {return_type} tmp_output;
       {return_type} reference_tensor_output;
       {{
-        at::AutoDispatchSkipFunctionalize guard;
+        // Exclude any modes: the purpose of calling into meta kernels is only as an implementation
+        // detail to perform shape inference, and we don't want any modal keys to run.
+        // Specifically, we want to prevent functionalization and Python modes from running.
+        c10::impl::ExcludeDispatchKeyGuard guard(
+            c10::functorch_transforms_ks |
+            c10::DispatchKeySet({{
```

Comment:
nit: factor the list of keys to skip out to reduce duplication and prevent them from going out of sync?